classes/robot/crawler.php

```diff
@@ -427,15 +429,20 @@ public function mark_for_crawl($baseurl, $url, $courseid = null, $priority = TOO
             }
 
             $node->id = $DB->insert_record('tool_crawler_url', $node);
-        } else if ( $node->needscrawl < self::get_config()->crawlstart ) {
-            // Push this node to the end of the queue.
-            $node->needscrawl = time();
+        } else {
 
-            if (isset($courseid)) {
-                $node->courseid = $courseid;
-            }
+            if ( $node->needscrawl < self::get_config()->crawlstart || $node->priority != $priority) {
+                // Push this node to the end of the queue.
+                $node->needscrawl = time();
```

Comment:
This logic feels wrong to me, why are we pushing an existing node that might get processed soonish, further back in the queue, if it is getting re-queued a second time? Perhaps with a higher priority the second time around? That seems counter productive.

It looks to me like the needscrawl and the priority logic should be independent

Reply:
I've separated the logic of the three different things that can lead us to update our item